addons/web/static/src/core/dropdown/dropdown_item.xml

```diff
@@ -12,7 +12,11 @@
       t-att-data-hotkey="props.hotkey"
       t-att="dataAttributes"
     >
-      <t t-slot="default" />
+      <span t-if="props.hasSwitch" class="form-switch d-flex justify-content-between align-items-center p-0">
+        <label for="switch_color_scheme"><t t-slot="default" /></label>
```

Comment:
This is a core component, meaning that this code will run for any dropdown-item...  it can't include elements' specific code.

Any switch generated using this system will have  `for="switch_color_scheme"`, the same applies for the `id`  in the following line of course.

We should generate these attributes somehow.

Reply:
Perhaps we could use the `id` to generate the switch id (by adding `_switch` at the end) ?
eg. 
``` js
return {
        type: "item",
        id: "theItemID",
        ...
```

``` xml
  <label for="theItemID_switch"><t t-slot="default" /></label>
  <input type="checkbox" id="theItemID_switch" class="form-check-input m-0" t-att-checked="props.isChecked"/>
```

What do you think ?

Reply:
Since the `<DropdownItem/>` has the priority of the click event, there's no need to add `id=""` and `for=""` on `<label/>` and `<input/>` so I removed it.

Reply:
> Since the <DropdownItem/> has the priority of the click event

What do you mean? In any case I think we should handle a scenario when multiple switches (checkboxes) are related.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#handling_multiple_checkboxes

Reply:
Fixed by using the `<Checkbox/>`component. 👍